scripts/zones/Metalworks/npcs/Darha.lua

```diff
@@ -1,11 +1,12 @@
 -----------------------------------
 -- Area: Metalworks
 -- NPC:  Darha
--- Type: Standard Info NPC
+-- Involved in quest: Bait and Switch
+-- !pos -77.922 2.000 21.201 237
 -----------------------------------
 package.loaded["scripts/zones/Metalworks/TextIDs"] = nil;
 -----------------------------------
-
+require("scripts/globals/baitandswitch");
```

Comment:
ok ya we definitely don't wanna mix up quest specific scripts in root globals folder, i'll have to think about how we reorganise these at some point

Reply:
that would also be a good idea so we can get the things like icanheararainbow.lua and all other "quest" type stuff into some kind of organized system

Reply:
@takhlaq maybe a questutils.lua to merge all the quest helper craps into? since looks like it'll be a bit large and we'd be making the existing quest global or the existing utils global kinda bloaty with just quest stuff

Reply:
im not so sure, was thinking more create a global quest table in quest.lua
then make a quest directory and a script with a table with functions and other properties for each quest (guess its easier to think of it class), like how kj proposed we do text ids
e.g.
`scripts/globals/quest.lua`
```lua
quests = quests or {}
quests.var_types = 
{
    charVar = 0,
    localVar = 1,
}
```
`scripts/globals/quest/bait_and_switch.lua`
```lua
quests = quests or {}
quests[logid].bait_and_switch =
{
    repeatable = false,
    vars = { main = "[Q<logid>]bait_and_switch", additional = { ["ass"] = quests.var_types.localVar } },
    check = function(player, npc, trade)
        -- think of something to check by npc name and zone later
    end,
    additional_funcs = { 
                         checkBaitProgress = function(player, npc)
                         end,
                         etc...  
                       }
}
```
just so it's easier to keep track of everything, instead of having to go through a fuckton of npc files
i havent fully thought this through yet, and any suggestions are welcome in discord before we start email spamming

Reply:
set it up so all the quests are like that and make that reference so i can have something to look at for changes... i'm not at the point where i can just create the most organized system without a reference yet. what's already there is the reference i used.

Reply:
ah it appears the last bit of my comment i lost when i copy/pasted the example out of editor
sorry, i didnt mean you need to make those changes - was just for those who are watching the repo to hop into discord and suggest changes we could make to how we handle quests
i'll edit that last part back in